doc/wol/Wake-on-LAN-HLD.md

```diff
@@ -0,0 +1,136 @@
+# Wake-on-LAN in SONiC
+
+## Table of Content
+
+- [Overview](#overview)
+- [Background](#background)
+- [Magic Packet](#magic-packet)
+- [CLI Design](#cli-design)
+- [gNOI Design](#gnoi-design)
+- [Test Plan](#test-plan)
+- [Reference](#reference)
+
+## Revision
+
+| Revision | Date       | Author     | Change Description |
+| -------- | ---------- | ---------- | ------------------ |
+| 1.0      | Nov 7 2023 | Zhijian Li | Initial proposal   |
+
+## Definitions/Abbreviations 
+
+| Abbreviation | Definition                                        |
+|--------------|---------------------------------------------------|
+| WoL          | **W**ake-**o**n-**L**AN                           |
+| gNOI         | **g**RPC **N**etwork **O**perations **I**nterface |
+| NIC          | **N**etwork **I**nterface **C**ontroller          |
+
+## Overview
+
+**Wake-on-LAN** (**WoL** or **WOL**) is an Ethernet or Token Ring computer networking standard that allows a computer to be turned on or awakened from sleep mode by a network message[^1]. This document describes the Wake-on-LAN feature design in SONiC.
+
+## Background
+
+Below diagram describes a common usage of WoL on SONiC switch, this HLD will focus on the green part:
+
+![WoL Background](./wol-background.png)
+
+## Magic Packet
+
+**Magic Packet** is a Ethernet frame with structure:
+
+* **Ethernet Frame Header**:
+  * **Destination MAC address**: broadcast MAC address (ff:ff:ff:ff:ff:ff) or target device's MAC address. [6 bytes]
+  * **Source MAC address**. [6 bytes]
+  * **EtherType**: `0x0842`. [2 bytes]
+* **Ethernet Frame Payload**:
+  * Six bytes of all `0xff`. [6 bytes]
+  * Sixteen repetitions of the target device's MAC address. [96 bytes]
+  * (Optional) A four or six byte password. [4 or 6 bytes]
+
+```
+Byte     |
+Offset   |0     |1     |2     |3     |4     |5     |
+---------+------+------+------+------+------+------+
+       0 |             Destination MAC             |
+         +------+------+------+------+------+------+        ETHERNET FRAME
+       6 |               Source MAC                |           HEADER
+         +------+------+---------------------------+
+      12 | 0x08 | 0x42 |   \      \      \      \
+         +------+------+------+------+------+------+------------------------------
+      14 | 0xFF | 0xFF | 0xFF | 0xFF | 0xFF | 0xFF |
+         +------+------+------+------+------+------+        ETHERNET FRAME
+      20 | Target MAC (repeat 16 times, 96 bytes)  |           PAYLOAD
+         +-----------------------------------------+
+     116 |    Password (optional, 4 or 6 bytes)    |
+         +-----------------------------------------+
+```
+
+## CLI Design
+
+The `wol` command is used to send magic packet to target device.
+
+This command requires root privilege because it'll modify the power-mode of target device.
```

Comment:
SONiC device just need to send the magic packet to target device(s), right? not sure why we need root privilege at  the SONiC device for command execution.

Reply:
Yes, SONiC device just send the magic packet. As the doc mentioned, the magic packet will modify the power-mode of the target device, so it's considered as an OPERATION command and requires root.

Reply:
Thanks for point this out, I think we can remove this root privilege requirement since SONiC only need to send a packet. I've update the doc.